packages/core/src/types/utils.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  Abi,
+  AbiFunction,
+  AbiParameter,
+  AbiParameterToPrimitiveType,
+  AbiParametersToPrimitiveTypes,
+  ExtractAbiFunction,
+} from 'abitype'
+
+export type IsNever<T> = [T] extends [never] ? true : false
+export type NotEqual<T, U> = [T] extends [U] ? false : true
+export type Or<T, U> = T extends true ? true : U extends true ? true : false
+
+export type GetArgs<
+  TAbi extends Abi | readonly unknown[],
+  TFunction extends AbiFunction & { type: 'function' },
+> = TFunction['inputs'] extends infer TInputs extends readonly any[]
+  ? Or<IsNever<TInputs>, NotEqual<TAbi, Abi>> extends true
+    ? {
+        /**
+         * Arguments to pass contract method
+         *
+         * Use a [const assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) on {@link contractInterface} for type inference.
+         */
+        args?: readonly any[]
```

Comment:
Should this be `any | readonly any[]` for backwards compat? Or is there a reason why it has to be an array?

Reply:
Yeah, fair call. Definitely prefer strictly passing an array!